### DISCUSSION TITLE
Cloudflare cycling on some sites when http proxy is used

### DISCUSSION THREAD

**Discussion by electrumguy** [seleniumbase/SeleniumBase#2055] [Q&A] [answered]
Using SeleniumBase (the driver manager method), I've noticed that when utilizing a http/https proxy, Cloudflare will just cycle. I'll get the turnstiles, solve them manually, then it will immediately "refresh" and ask me to validate again. It will not tell me my solution is wrong, it'll just cycle.

Running SeleniumBase without the http/https proxy will not generate this behavior. Does SeleniumBase handle http/https proxies differently which is why they are being "detected"? Is this because I am utilizing the driver manager method documented here: https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_23 as opposed to BaseClass? 

I understand more than likely the http/https proxy itself is the issue, would it be fair to assume if I get the manual Cloudflare prompt, the IP has been blacklisted by Cloudflare? Is there a way to validate this? Any insight would be very appreciated!

**Comment by mdmintz** [comment, marked as answer]
Are you setting your proxy with ``proxy="server:port"`` or `proxy="user:pass@server:port"`?

There's also the pytest command-line option way:

```bash
pytest --proxy=server:port

pytest --proxy=user:pass@server:port
```

That's the official SeleniumBase way of making it work.

**Comment by electrumguy** [comment]
Im setting it via: proxy=“server:port” when declaring the DriverOn Sep 2, 2023, at 11:40 AM, Michael Mintz ***@***.***> wrote:﻿
Are you setting your proxy with proxy="server:port" or ***@***.***:port"?
There's also the pytest command-line option way:
pytest --proxy=server:port

pytest ***@***.***:port
That's the official SeleniumBase way of making it work.

**Reply by mdmintz** [reply]
If CF has blacklisted an IP, expect to get blocked. Otherwise, changing proxy settings works.

**Reply by electrumguy** [reply]
When you say "blocked" would I be getting a prompt on the page specifying such? I'm not entirely sure what you mean by changing proxy settings, are you referring to changing the proxy itself to a different one?

**Reply by mdmintz** [reply]
They don't tell you when you're blocked. You just get the turnstile / captcha. They can blacklist an IP Address, meaning that you'll get blocked unless you change proxy settings.
